sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/EmbeddedMetastoreService.java

```diff
@@ -0,0 +1,120 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.hcatalog;
+
+import java.io.File;
+
+import org.apache.hadoop.hive.cli.CliSessionState;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.ql.CommandNeedRetryException;
+import org.apache.hadoop.hive.ql.Driver;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hive.hcatalog.common.HCatUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of a light-weight embedded metastore.
+ * This class is a trimmed-down version of <a href="https://github.com/apache/hive/blob/master
+/hcatalog/core/src/test/java/org/apache/hive/hcatalog/mapreduce/HCatBaseTest.java">
+ * https://github.com/apache/hive/blob/master/hcatalog/core/src/test/java/org/apache/hive/hcatalog/mapreduce
+ * /HCatBaseTest.java </a>
+ */
+public class EmbeddedMetastoreService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(EmbeddedMetastoreService.class);
+  private static final String BASE_DIR = System.getProperty("user.dir");
+  private static final String HIVE_DIR = "/target/hive";
+  private static final String TEST_DATA_DIR = HCatUtil.makePathASafeFileName(BASE_DIR + HIVE_DIR
+      + "/data/" + EmbeddedMetastoreService.class.getCanonicalName() + "-"
+      + System.currentTimeMillis());
+  private static final String TEST_WAREHOUSE_DIR = TEST_DATA_DIR + "/warehouse";
+  private static final EmbeddedMetastoreService SVC_INSTANCE = new EmbeddedMetastoreService();
+
+  static {
+    try {
+      initiateService();
+    } catch (MetaException e) {
+      LOG.error("Exception while initiating service", e);
+    }
+  }
+
+  private Driver driver;
+  private HiveConf hiveConf;
+
+  private EmbeddedMetastoreService() {
+  }
+
+  private static void initiateService() throws MetaException {
+
+    //set-up the directories for test datastore
+    setUpFSDirectory(HIVE_DIR);
+    setUpFSDirectory(TEST_DATA_DIR);
+    setUpFSDirectory(TEST_WAREHOUSE_DIR);
+
+    //below properties are used by theembedded metastore to store config
+    //and data files during execution
+    System.setProperty("test.tmp.dir", HCatUtil.makePathASafeFileName(
```

Comment:
This seems icky to do in a `static` block. What if another test also uses `test.tmp.dir`?